### DISCUSSION TITLE
Example of using Jaeger with Prometheus and Grafana

### DISCUSSION THREAD

**Discussion by zhylinskiy-stryber** [jaegertracing/jaeger#2762] [Q&A]
Hello everyone.
My goal is to set up monitoring for my app, make alerts for pretty standard cases, like very long http request, or heavy SQL query.
I have an app written in PHP using Laravel framework. I got to the point where I can send traces to Jaeger with proper information attached, that works fine.
I run my environment using docker-compose. I have Jaeger all-in-one, Prometheus and Grafana. I managed to scrape metrics from Jaeger, using Prometheus. Little off topic: Would be nice if someone pointed me out, why cannot Grafana communicate directly to Jaeger, and instead uses Prometheus.
I copied default Grafana dashboard from repo and it didn't work for 100%, only few graphics work.
I also tried the way of using jb and jsonnet to generate dashboard JSON, but still it doesn't work for 100%

Please,
can anyone provide any useful information on how to:
1. Get Grafana dashboard work.
2. How can I expose traces log information (the one I send from my app) and utilise it using Prometheus and Grafana.

Thank you.

**Comment by jpkrohling** [comment]
Are you talking about this one here? https://github.com/jaegertracing/jaeger/tree/master/monitoring/jaeger-mixin

What does "didn't work 100% mean"? Did you see error messages?

**Reply by zhylinskiy-stryber** [reply]
It looks like this, some metrics are indeed fetched, but overall "No data"
![image](https://user-images.githubusercontent.com/61143234/106618095-d2f95000-6577-11eb-8ffe-77211634fa3e.png)

**Reply by zhylinskiy-stryber** [reply]
whereas, Prometheus gets all the metrics, just an example
![image](https://user-images.githubusercontent.com/61143234/106618735-78142880-6578-11eb-8137-9e7a0dd2ccd4.png)

**Reply by zhylinskiy-stryber** [reply]
And this is what I have in Jaeger:
![image](https://user-images.githubusercontent.com/61143234/106618832-97ab5100-6578-11eb-8754-eb3b4693ea43.png)

**Reply by jpkrohling** [reply]
If you compare what you get from Jaeger's own `/metrics` endpoint, does it match what you are seeing in Prometheus?

**Reply by zhylinskiy-stryber** [reply]
No, Prometheus  `/metrics` does not expose Jaeger's metrics, thanks for pointing out, is there a way to expose them ?

**Reply by jpkrohling** [reply]
I'm not sure we are talking about the same thing: Prometheus' own metrics won't show Jaeger metrics. Prometheus should be scrapping Jaeger metrics, making them available in the query. If you get Jaeger's `/metrics`, your goal should be to query them in Prometheus. If that's not happening, you'll have to check your Prometheus scrape jobs under `Targets` in your Prometheus UI.

**Reply by zhylinskiy-stryber** [reply]
Ah yes, I got your point. I have this in my prometheus.yml config file

![image](https://user-images.githubusercontent.com/61143234/106735998-95e79900-661d-11eb-9b81-269766d1c653.png)

      
It works fine: 
![image](https://user-images.githubusercontent.com/61143234/106734602-e231d980-661b-11eb-95d4-eac7ada1373b.png)

**Reply by jpkrohling** [reply]
It means that all metrics available to monitor Jaeger are getting scraped by Prometheus. Are you by any chance expecting to see metrics related to your workload instead of internal Jaeger metrics? If so, then see the answer by @jkowall.

**Reply by zhylinskiy-stryber** [reply]
Alright, thanks.

**Comment by jkowall** [comment]
Looks like metric aggregation and alerting. @zhylinskiy-stryber can you have a look at this and see if this describes what you want to do? https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/403#issuecomment-742257626

**Reply by zhylinskiy-stryber** [reply]
So you suggest using Opentelemtry-collector instead of Jaeger ?
